### PR TITLE
Simplify C compilation for Fortanix-SGX target

### PR DESCRIPTION
**Pull request by jethrogb**
cc @raoulstrackx

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -7,10 +7,11 @@
...
 src/etc/installer/gfx/* binary
-*.woff binary
 src/vendor/** -text
 Cargo.lock linguist-generated=false
 
 # Older git versions try to fix line endings on images, this prevents it.
 *.png binary
 *.ico binary
+*.woff binary
+*.woff2 binary
```
Comment (joshtriplett) [inline .gitattributes:17]: This seems like an unrelated change bundled into this PR.

File: src/ci/docker/host-x86_64/dist-various-2/Dockerfile [hunk 2]
```diff
@@ -45,10 +45,10 @@ ENV \
...
     CXX_armv7_unknown_linux_gnueabi=arm-linux-gnueabi-g++-8 \
     AR_x86_64_fortanix_unknown_sgx=ar \
-    CC_x86_64_fortanix_unknown_sgx=x86_64-fortanix-unknown-sgx-clang-11 \
-    CFLAGS_x86_64_fortanix_unknown_sgx="-mlvi-hardening -mllvm -x86-experimental-lvi-inline-asm-hardening" \
-    CXX_x86_64_fortanix_unknown_sgx=x86_64-fortanix-unknown-sgx-clang++-11 \
-    CXXFLAGS_x86_64_fortanix_unknown_sgx="-mlvi-hardening -mllvm -x86-experimental-lvi-inline-asm-hardening" \
+    CC_x86_64_fortanix_unknown_sgx=clang-11 \
+    CFLAGS_x86_64_fortanix_unknown_sgx="-D__ELF__ -isystem/usr/include/x86_64-linux-gnu -mlvi-hardening -mllvm -x86-experimental-lvi-inline-asm-hardening" \
+    CXX_x86_64_fortanix_unknown_sgx=clang++-11 \
+    CXXFLAGS_x86_64_fortanix_unknown_sgx="-D__ELF__ -isystem/usr/include/x86_64-linux-gnu -mlvi-hardening -mllvm -x86-experimental-lvi-inline-asm-hardening" \
```
Comment (joshtriplett) [inline src/ci/docker/host-x86_64/dist-various-2/Dockerfile:51]: Interesting. In general, I'd be concerned about assuming that system headers will work for a different target (it's always a challenge to build a non-host target using a host compiler while cancelling out any assumptions that hold for one and not the other), but I don't think that's an issue for a highly controlled Docker environment like this.

**Comment by joshtriplett** [comment]
Looks reasonable to me. r=me if you split out the unrelated change into a separate PR.

File: src/ci/docker/host-x86_64/dist-various-2/Dockerfile (hunk 2 above)
Comment (jethrogb) [reply src/ci/docker/host-x86_64/dist-various-2/Dockerfile:51]: Not just “system headers” but specifically the `x86_64-linux-gnu` headers. I think it's better this way than the previous solution.

**Comment by jethrogb** [comment]
@joshtriplett Thanks, can you delegate to @raoulstrackx I'd like him to take a look as well.

**Comment by raoulstrackx** [comment]
bors r=me
